lib/utils/index.js

```diff
@@ -0,0 +1,20 @@
+var path = require('path')
+  , tmpdir = require('os').tmpdir()
+  , fs = require('fs')
+  , mkdirp = require('mkdirp')
+  , os = require('os')
+  , uuid = require('uuid');
+
+function tempFilePath(filepath) {
+  return path.join(filepath || tmpdir, uuid.v1());
+}
+
+module.exports.writeTempFile = function (str) {
+  var isWin = /^win/i.test(os.platform());
+  var fullpath = isWin ? tempFilePath('.') : tempFilePath();
+
+  mkdirp.sync(path.dirname(fullpath));
```

Comment:
I think it's not necessary to create folders at either `.` or whatever `require('os').tmpdir()` returns, as they already exist.